### ISSUE TITLE
Crash & kernel panic when resizing a terminal to "very small"

### ISSUE THREAD

**Issue by neonrust** [SerenityOS/serenity#11748]
I just tried resizing the terminal that opens on boot (works with any terminal, though), to a minimal size, basically to see if it would cause problems. It did :)  As the terminal crashes, the kernel panics.

I assume this must be some env problem, because must've been seen by someone else at some point...
First time I've tried Serenity, I'm on Ubuntu 21.10 and I used the distro's qemu.

This was dumped in the boot terminal:
```
18.631 Shell(31:31): ASSERTION FAILED: TODO
../../Userland/Libraries/LibLine/Editor.cpp:203
18.631 [#0 Shell(31:31)]: Terminating Shell(31) due to signal 6
18.634 [#0 FinalizerTask(4:4)]: 0xdeadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) + 0x31e
0xdeadc0de  Kernel::Scheduler::context_switch(Kernel::Thread*) [clone .localalias] + 0x222
0xdeadc0de  Kernel::Scheduler::pick_next() [clone .localalias] + 0x1df
0xdeadc0de  Kernel::Processor::check_invoke_scheduler() [clone .localalias] + 0x175
0xdeadc0de  Kernel::Thread::die_if_needed() [clone .localalias] + 0x40d
0xdeadc0de  syscall_handler + 0x1d42
0xdeadc0de  syscall_asm_entry + 0x31
0x4a101825  libc.so: .text + 0x16825
0x4a0eca7a  libc.so: .text + 0x1a7a
0x4a0ed50d  libc.so: .text + 0x250d
0x6530c504  libline.so.serenity: .text + 0x3504
0x65310937  libline.so.serenity: .text + 0x7937
0x65310ce8  libline.so.serenity: .text + 0x7ce8
0x653110cb  libline.so.serenity: .text + 0x80cb
0x3c1cf51b  libcore.so.serenity: .text + 0x1751b
0x3c1cfa4d  libcore.so.serenity: .text + 0x17a4d
0x3c1cfe3d  libcore.so.serenity: .text + 0x17e3d
0x3c1d0648  libcore.so.serenity: .text + 0x18648
0x3c1d133b  libcore.so.serenity: .text + 0x1933b
0x6531c8ed  libline.so.serenity: .text + 0x138ed
0x2d64e994  libshell.so.serenity: .text + 0x7f994
0x2d64ecb9  libshell.so.serenity: .text + 0x7fcb9
0x3c1e61a1  libcore.so.serenity: .text + 0x2e1a1
0x3c1d078f  libcore.so.serenity: .text + 0x1878f
0x3c1d133b  libcore.so.serenity: .text + 0x1933b
0xa6ee21b3  /bin/Shell: .text + 0x11b3
0xa6ee23dd  /bin/Shell: .text + 0x13dd

18.650 [#0 FinalizerTask(4:4)]: Generating coredump for pid: 31
18.675 CrashDaemon(9:9): New coredump file: /tmp/coredump/Shell_31_1641692730
18.737 CrashReporter(35:35): Started thread "", tid = 36
18.745 Terminal(28:28): TerminalWidget: EOF on master pty, firing on_command_exit hook.
[Terminal(28:28)]: ASSERTION FAILED: m_state == State::BeingRemoved
[Terminal(28:28)]: ../../Kernel/Devices/Device.cpp:146 in virtual Kernel::Device::~Device()
[Terminal(28:28)]: KERNEL PANIC! :^(
[Terminal(28:28)]: Aborted
[Terminal(28:28)]: at ../../Kernel/Arch/x86/common/CPU.cpp:35 in void abort()
[Terminal(28:28)]: Kernel + 0x00c742b1  Kernel::__panic(char const*, unsigned int, char const*) +0xf1
[Terminal(28:28)]: Kernel + 0x0101e8b9  abort.localalias +0x245
[Terminal(28:28)]: Kernel + 0x0101e674  abort.localalias +0x0
[Terminal(28:28)]: Kernel + 0x00529d02  Kernel::Device::~Device() [clone .localalias] +0x6c2
[Terminal(28:28)]: Kernel + 0x0051e144  Kernel::CharacterDevice::~CharacterDevice() +0x54
[Terminal(28:28)]: Kernel + 0x00ef1591  Kernel::MasterPTY::~MasterPTY() [clone .localalias] +0x381
[Terminal(28:28)]: Kernel + 0x00ef18cc  Kernel::MasterPTY::~MasterPTY() [clone .localalias] +0x1c
[Terminal(28:28)]: Kernel + 0x008be3bb  Kernel::OpenFileDescription::~OpenFileDescription() [clone .localalias] +0xd7b
[Terminal(28:28)]: Kernel + 0x00e1646b  Kernel::Process::sys$close(int) +0xa3b
[Terminal(28:28)]: Kernel + 0x00d384d7  syscall_handler +0x1f97
[Terminal(28:28)]: Kernel + 0x00d36531  syscall_asm_entry +0x31
```
And the screen looked like this:
![image](https://user-images.githubusercontent.com/6277212/148665930-76e05195-671e-46da-9392-09cb5ad5f7da.png)

**Comment by alimpfard** [comment]
The LibLine issue is well-known (it's actually an unimplemented behaviour: "what do when prompt doesn't fit?")
See the...multitude of issues if you search for "terminal resize".

The kernel panic is probably new though (I haven't seen it before at least)

**Comment by BertalanD** [comment]
The kernel panic is likely the `RefPtr` regression that was fixed in 5871072

**Comment by neonrust** [comment]
Ok yeah, updated and tried again, and now it doesn't happen any more.
